### PR TITLE
DELTA-894: Gutenberg Support

### PR DESCRIPTION
**Pull request by renatonascalves**
Making sure the Clone and Replace functionality works with Gutenberg.

### CODE REVIEW

File: config/stylelint.config.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    'selector-combinator-space-before': 'always',
+    'selector-descendant-combinator-no-non-space': true,
+    'selector-list-comma-newline-after': 'always',
+    'selector-max-specificity': [
+      '0,2,1',
+      {
+        ignoreSelectors: [
+          ':global',
+          ':local',
+          '#oovvuu-dialog-wrapper',
```
Comment (kjbenk) [inline config/stylelint.config.js:56]: Please remove these references

File: config/stylelint.config.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          '#oovvuu-alert-wrapper',
```
Comment (kjbenk) [inline config/stylelint.config.js:57]: same as above

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,13 @@
...
+.DS_Store
+.env
 .svn
 .vscode
+
+# Client files
+node_modules
+
+# Built files
+vendor
```
Comment (kjbenk) [inline .gitignore:12]: Do we need the vendor files here since this is meant to be a built version of the plugin?

File: class-cr-search.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
+		$current_post_ud = absint( $request->get_param( 'current_post_id' ) );
```
Comment (kjbenk) [inline class-cr-search.php:32]: Fix spelling to `current_post_id` as well as the reference below

File: class-cr-search.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+		$query_args = array(
+			'post__not_in'        => array( $current_post_ud ), // phpcs:ignore WordPressVIPMinimum.Performance.WPQueryParams.PostNotIn
+			'post_type'           => get_post_type( $current_post_ud ),
+			'post_status'         => 'publish',
+			'paged'               => absint( $request->get_param( 'page' ) ),
+			'posts_per_page'      => absint( $request->get_param( 'per_page' ) ),
```
Comment (kjbenk) [inline class-cr-search.php:39]: Can you set a max value of 100 for this?

```php
min( 100, $request->get_param( 'per_page' ) )
```

File: package.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    "eslint-import-resolver-babel-module": "^5.1.2",
+    "eslint-loader": "^3.0.3",
+    "eslint-plugin-import": "^2.20.2",
+    "eslint-plugin-jsx-a11y": "^6.1.1",
+    "eslint-plugin-react": "^7.19.0",
+    "file-loader": "^6.0.0",
+    "identity-obj-proxy": "^3.0.0",
+    "react-svg-loader": "^3.0.3",
+    "resolve-url-loader": "^3.1.1",
+    "url-loader": "^4.1.0",
```
Comment (kjbenk) [inline package.json:54]: Do we need these packages?

File: readme.txt [hunk 7]
```diff
@@ -3,7 +3,7 @@ Contributors: mboynes, alleyinteractive
 Tags: editing, duplication, replacement, workflow
 Requires at least: 3.5
 Tested up to: 3.6
```
Comment (kjbenk) [inline readme.txt:5]: Please update to 5.5 if that is what you tested it against :)

File: readme.txt [hunk 8]
```diff
@@ -1,8 +1,8 @@
 === Plugin Name ===
 Contributors: mboynes, alleyinteractive
 Tags: editing, duplication, replacement, workflow
-Requires at least: 3.5
-Tested up to: 3.6
+Requires at least: 4.9
```
Comment (kjbenk) [inline readme.txt:4]: Is there any reason why we cannot support back to 3.5?

Comment (renatonascalves) [reply readme.txt]: Ops!

File: clone-replace.php [hunk 9]
```diff
@@ -32,10 +32,35 @@
 add_action(
 	'init',
 	function () {
+		if ( function_exists( 'register_post_meta' ) ) {
+			register_post_meta(
+				'post',
```
Comment (mslinnea) [inline clone-replace.php:37]: Does this plugin work with custom post types?  Should this be for all post types that are supported?  Same question for the next register_post_meta call.

Comment (renatonascalves) [reply clone-replace.php:37]: The current implementation for the classic editor has no support for other post types so I decided to follow up here.

**Comment by srtfisher** [comment]
@renatonascalves are we good to merge this on in?

**Comment by renatonascalves** [comment]
@srtfisher Code-wise, yes, but I'm still waiting UAT from the stakeholder.

**Comment by montchr** [comment]
@renatonascalves @justinwhall Is this pull request still necessary considering #21 has been merged?

**Comment by renatonascalves** [comment]
I think someone added this feature in another pull request. I don't remember who now.

**Comment by montchr** [comment]
Gutenberg support was added in #21.

**Pull request closed without merging**